mesh/mesh.hpp

```diff
@@ -990,6 +990,11 @@ class Mesh
        for example, after the mesh nodes are modified externally. */
    void DeleteGeometricFactors();
 
+   /// This function should be called after the mesh node coordinates have
+   /// changed, e.g. after the mesh has moved. It updates internal quantities
+   /// derived from the node coordinates, suchs as the GeometricFactors.
+   void Update() { DeleteGeometricFactors(); }
```

Comment:
Maybe `Update` is too general/generic? It doesn't need to be called after e.g. uniform or local refinements, because those refinements will internally call `ResetLazyData`.

Perhaps `NodesHaveMoved` or `MeshHasMoved` would be more self-explanatory?

Reply:
I thought about it and couldn't come up with a better name... I am very open to suggestions 😄 

I didn't like nodes/moved because this really should be called any time the mesh (i.e. node coordinates and maybe mesh topology) has changed. But maybe I am overthinking it...

What do other people think?

Reply:
Personally I like `Update` because this is already used for the AMR related changes and I won't need to remember a new function name. It is general but, if needed in the future, we could add more specific `Update*` methods to provide users with more fine-grained control.

Reply:
Isn't it confusing though that you **should** call `FiniteElementSpace::Update` after mesh refinement, but **should not** call `Mesh::Update` after refinement?

Reply:
I think this is where finer control would be useful. Would there be any harm in calling `Mesh::Update` after refinement (aside from potential performance penalties)? If not then the naive user could call `Update` on virtually everything after any type of mesh change; topological or geometrical. The more advanced user could update only what is necessary after a particular type of change. As long as we have sufficient documentation so that the more advanced user can make intelligent choices.